x/profile/client/cli/tx.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/desmos-labs/desmos/x/profile/internal/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// GetTxCmd set the tx commands
+func GetTxCmd(_ string, cdc *codec.Codec) *cobra.Command {
+	postsTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Posts transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	postsTxCmd.AddCommand(flags.PostCommands(
+		GetCmdCreateProfile(cdc),
+		GetCmdEditProfile(cdc),
+		GetCmdDeleteProfile(cdc),
+	)...)
+
+	return postsTxCmd
+}
+
+// GetCmdCreateProfile is the CLI command for creating a profile
+func GetCmdCreateProfile(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [moniker]",
+		Short: "Create a new profile",
+		Long: fmt.Sprintf(`
+Create a new profile specifying the moniker, name, surname, bio, a profile picture and cover.
+Every data except moniker are optional, let be free to specify only what you want to.
+
+E.g (only with moniker)
+%s tx profile create leoDiCap 
+
+E.g (with all the other optional fields)
+%s tx profile create leoDiCap \
+	--name "Leonardo" \
+	--surname "Di Caprio" \
+	--bio "Hollywood actor. Proud environmentalist" \
+	--picture "https://profilePic.jpg"
+	--cover "https://profileCover.jpg"
+`, version.ClientName, version.ClientName),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			picture := viper.GetString(flagProfilePic)
+			cover := viper.GetString(flagProfileCover)
+			pictures := types.NewPictures(picture, cover)
+
+			name := viper.GetString(flagName)
+			surname := viper.GetString(flagSurname)
+			bio := viper.GetString(flagBio)
+
+			msg := types.NewMsgCreateProfile(name, surname, args[0], bio, &pictures, cliCtx.FromAddress)
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+
+	cmd.Flags().String(flagName, "", "Name of the profile")
+	cmd.Flags().String(flagSurname, "", "Surname of the profile")
+	cmd.Flags().String(flagBio, "", "Biography of the profile")
+	cmd.Flags().String(flagProfilePic, "", "Profile related profile picture")
+	cmd.Flags().String(flagProfileCover, "", "Profile related profile cover")
+
+	return cmd
+}
+
+// GetCmdEditProfile is the CLI command for editing an profile
+func GetCmdEditProfile(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit [previous_moniker] [[new_moniker]]",
```

Comment:
I think that in order to make this command more similar to the creation, we should have the new moniker specified as a flag. So that if the user wants to edit he has to specify `--moniker = `. 

Also, do we really need him to specify his current moniker? Can't we simply get his current profile based on his address?

Reply:
Yes we could definitely do this using the address